tests/phpunit/Configuration/Entry/LogsAssertions.php

```diff
@@ -65,7 +63,8 @@ private function assertLogsStateIs(
             $this->assertNull($badge);
         } else {
             $this->assertNotNull($badge);
-            $this->assertBadgeStateIs($badge, $expectedBadge->getBranch());
```

Comment:
Removed this: it made the tests much more confusing than they were, and a simpler solution was to do object-to-object equality.

Reply:
object-to-object equality is always tricky due to the loose comparison, but if anyone is interested there is https://github.com/webmozarts/strict-phpunit which solves that issue and allows to remove a lot of boilertemplate